### ISSUE TITLE
Decimal arithmetic problems

### ISSUE THREAD

**Issue by Kledal** [sarahdayan/dinero.js#6]
```js
let added = Dinero({ amount: 0.1 }).add(Dinero({ amount: 0.2 }))
let result = added.equalsTo(Dinero({ amount: 0.3 }))
```
`result` is false! I can't believe there's a currency library that doesn't do correct decimal arithmetic.

https://codepen.io/recursive/pen/VXogvN?editors=0010

HackerNews comment: https://news.ycombinator.com/item?id=16852357

**Comment by montanaflynn** [comment]
I blame Javascript for this:

```js
let added = Dinero({ amount: 0.1 }).add(Dinero({ amount: 0.2 }))
console.log("dinero", added.getAmount())

let addedVanilla = 0.1 + 0.2;
console.log("vanilla", addedVanilla)

// "dinero"  0.30000000000000004
// "vanilla" 0.30000000000000004
```

**Comment by sarahdayan** [comment]
@Kledal @montanaflynn Amount is meant to be expressed in cents, with integers, [as the documentation suggests it](https://sarahdayan.github.io/dinero.js/module-Dinero.html). But I agree, this isn't explicit enough and the library should throw when a non-integer is passed as an amount.

This will be fixed.

**Comment by sarahdayan** [comment]
Issue fixed in [v1.0.0](https://github.com/sarahdayan/dinero.js/releases/tag/v1.0.0)

**Comment by sarahdayan** [comment]
@m5r It's a good question. Floats should be ruled out anyway so storing them "as is" isn't an option.

It seems to me that the best option is to round using [Banker's rounding](http://wiki.c2.com/?BankersRounding) before creating a new Dinero object in the `divide` and `multiply` methods.
